### PR TITLE
.NET: Treat warnings as errors

### PR DESCRIPTION
**Pull request by justinvp**
Fix the following warning and turn on "treat warnings as errors" for our .NET projects.

```
Deployment/InvokeOptions.cs(6,60): warning CS1574: XML comment has cref attribute 'InvokeAsync{T}(string,
```

### CODE REVIEW

File: sdk/dotnet/Pulumi/Pulumi.csproj [hunk 1]
```diff
@@ -9,14 +11,16 @@
     <RepositoryUrl>https://github.com/pulumi/pulumi</RepositoryUrl>
     <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
     <PackageIcon>pulumi_logo_64x64.png</PackageIcon>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+  </PropertyGroup>
 
-    <TargetFramework>netcoreapp3.0</TargetFramework>
-    <Nullable>enable</Nullable>
+  <PropertyGroup>
+    <NoWarn>NU5105</NoWarn>
```
Comment (mikhailshilkov) [inline sdk/dotnet/Pulumi/Pulumi.csproj]: Curious why this change?

Comment (justinvp) [reply sdk/dotnet/Pulumi/Pulumi.csproj]: I wasn't sure why we were only setting it for `Debug|AnyCPU`. This sets it for both Release and Debug configurations like `Pulumi.FSharp`.

Comment (justinvp) [reply sdk/dotnet/Pulumi/Pulumi.csproj]: Do we even do Release builds?

Comment (mikhailshilkov) [reply sdk/dotnet/Pulumi/Pulumi.csproj]: No :)

Comment (justinvp) [reply sdk/dotnet/Pulumi/Pulumi.csproj]: > No :)

By design? I'm happy to revert this part of the change if it's unnecessary.

Comment (mikhailshilkov) [reply sdk/dotnet/Pulumi/Pulumi.csproj]: I think we just never discussed this. I suspect we don't want to use semver 2.0 for release versions but I may be wrong.
